src/hooks/use-certificate-trust.ts

```diff
@@ -0,0 +1,42 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useWindowListener } from 'src/hooks/use-window-listener';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+/**
+ * Custom hook that checks if the Studio CA certificate is trusted on the system
+ * @returns A boolean indicating if the certificate is trusted
+ */
+export function useCertificateTrust(): boolean {
+	const isMounted = useRef( true );
+	const [ isTrusted, setIsTrusted ] = useState< boolean >( false );
+
+	const checkCertificateTrust = useCallback( () => {
+		// If the certificate is already trusted, don't check it again
+		if ( isTrusted ) {
+			return;
+		}
+
+		getIpcApi()
+			.isCATrusted()
+			.then( ( trusted ) => {
+				if ( isMounted.current ) {
+					setIsTrusted( trusted );
+				}
+			} )
+			.catch( ( error ) => {
+				console.error( 'Failed to check certificate trust:', error );
+			} );
+	}, [ setIsTrusted, isTrusted ] );
+
+	useWindowListener( 'focus', checkCertificateTrust );
+
+	useEffect( () => {
+		isMounted.current = true;
+		checkCertificateTrust();
+		return () => {
+			isMounted.current = false;
+		};
+	}, [ checkCertificateTrust ] );
+
+	return isTrusted;
+}
```

Comment:
```suggestion
export function useCertificateTrust(): boolean {
	const [ isTrusted, setIsTrusted ] = useState< boolean >( false );

	const checkCertificateTrust = useCallback( () => {
		// If the certificate is already trusted, don't check it again
		if ( isTrusted ) {
			return;
		}

		getIpcApi()
			.isCATrusted()
			.then( ( trusted ) => {
				setIsTrusted( trusted );
			} )
			.catch( ( error ) => {
				console.error( 'Failed to check certificate trust:', error );
			} );
	}, [ setIsTrusted, isTrusted ] );

	useWindowListener( 'focus', checkCertificateTrust );

	useEffect( () => {
		checkCertificateTrust();
	}, [ checkCertificateTrust ] );

	return isTrusted;
}
```

Condsider this a suggestion, but after testing this a bit, I don't think using a ref here prevents any extra executions. Was it a different purpose this was added for?

Reply:
Theoretically, the component could very well have unmounted between the `checkCertificateTrust` call and when it resolves, and updating the state of an unmounted component will generate a warning.

Reply:
I am considering refactoring `useCertificateTrust` using Redux to avoid managing local states.

Reply:
I created follow up PR: https://github.com/Automattic/studio/pull/1325